src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1150 @@
+require 'tempfile'
+require 'fileutils'
+require 'date'
+
+module Auth
+    class AuthConf
+        include Yast
+        include I18n
+        include Logger
+        include UIShortcuts
+
+        # Return network host name and environment facts.
+        def self.get_net_facts
+            resolvable_name = ''
+            begin
+                resolvable_name = Socket.gethostbyname(Socket.gethostname)
+                resolvable_name = resolvable_name.fetch(0, '')
+            rescue
+                # Intentionally ignored
+            end
+            ip_addresses = []
+            begin
+                resolvable_ip = Socket.getaddrinfo(Socket.gethostname, nil)
+                resolvable_ip = resolvable_ip.fetch(0, []).fetch(3, '')
+                if resolvable_ip != ''
+                    ip_addresses = [resolvable_ip]
+                end
+            rescue
+                # Just get interface IPs
+                ip_addresses = Socket.getifaddrs.select{|iface| iface.addr.ip?}.map{|iface| iface.addr.ip_address}.select{|addr| !addr.start_with?('127.')}
+            end
+            domain_name, status = Open3.capture2('domainname')
+            if status.exitstatus != 0 || domain_name.strip == '(none)'
+                domain_name = ''
+            end
+            domain_name.strip!
+            facts = {
+                'computer_name' => Socket.gethostname,
+                'full_computer_name' => resolvable_name,
+                'network_domain' => domain_name.chomp,
+                'ip_addresses' => ip_addresses,
+            }
+            return facts
+        end
+
+        SSSDCapableNSSDatabases = ["passwd", "group", "sudoers", "automount"]
+        LDAPCapableNSSDatabases = ["passwd", "group", "sudoers", "automount"]
+
+        # Enable the specified NSS database.
+        def nss_enable_module(db_name, module_name)
+            names = Nsswitch.ReadDb(db_name)
+            if !names.include?(module_name)
+                Nsswitch.WriteDb(db_name, names + [module_name])
+                Nsswitch.Write
+            end
+        end
+
+        # Disable the specified NSS database.
+        def nss_disable_module(db_name, module_name)
+            names = Nsswitch.ReadDb(db_name)
+            if names.include?(module_name)
+                Nsswitch.WriteDb(db_name, names.select { |name| name != module_name })
+                Nsswitch.Write
+            end
+        end
+
+        # Enable and start the service.
+        def service_enable_start(name)
+            Service.Enable(name)
+            if !(Service.Active(name) ? Service.Restart(name) : Service.Start(name))
+                Report.Error(_('Failed to start service %s. Please use system journal (journalctl -n -u %s) to diagnose.') % [name, name])
+            end
+        end
+
+        # Disable and stop the service.
+        def service_disable_stop(name)
+            Service.Disable(name)
+            if Service.Active(name) && !Service.Stop(name)
```

Comment:
I don't get it. No matter service is stopped or not, Service.Stop always returns true if the service is stopped, what's wrong the logic?